internal/factsengine/gatherers/verifypassword.go

```diff
@@ -0,0 +1,94 @@
+package gatherers
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	crypt "github.com/tredoe/osutil/user/crypt"
+	sha512crypt "github.com/tredoe/osutil/user/crypt/sha512_crypt"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const (
+	VerifyPasswordGathererName = "verify_password"
+	checkableUsernames         = "hacluster"
+	unsafePasswords            = "linux"
+)
+
+// nolint:gochecknoglobals
+var (
+	VerifyPasswordInvalidUsername = entities.FactGatheringError{
+		Type:    "verify-password-invalid-username",
+		Message: "requested user is not whitelisted for password check",
+	}
+)
+
+type VerifyPasswordGatherer struct {
+	executor utils.CommandExecutor
+}
+
+func NewDefaultPasswordGatherer() *VerifyPasswordGatherer {
+	return NewVerifyPasswordGatherer(utils.Executor{})
+}
+
+func NewVerifyPasswordGatherer(executor utils.CommandExecutor) *VerifyPasswordGatherer {
+	return &VerifyPasswordGatherer{
+		executor,
+	}
+}
+
+/*
+This gatherer expects only the username for which the password will be verified
+*/
+func (g *VerifyPasswordGatherer) Gather(factsRequests []entities.FactRequest) ([]entities.Fact, error) {
+	facts := []entities.Fact{}
+	log.Infof("Starting password verifying facts gathering process")
+
+	for _, factReq := range factsRequests {
+		if !strings.Contains(checkableUsernames, factReq.Argument) {
```

Comment:
this is prone to error, the string "ha" is contained in the string "hacluster,something"
why don't we use a slice, and check for the existence in the array?